### ISSUE TITLE
Verify that unreferenced LOIs and submissions won't break the App

### ISSUE THREAD

**Issue by DaoyuT** [google/ground-android#2279]
**Describe the bug**
In the web app, when user is deleting jobs, it also deletes its associated LOIs and submissions. However, we couldn't make all these deletions in one transaction because the [limitation](https://firebase.google.com/docs/firestore/manage-data/transactions#security_rules_limits) of accessing 20 docs in one transaction. So those calls were made asynchronously and there might be unreferenced LOIs and submissions. I filed a [bug](https://github.com/google/ground-platform/issues/1532) in the web app to clean it up periodically, and this bug is to make sure that such situation won't break the Android App.

**Comment by jcqli** [comment]
Task: delete the job but not the submission or LOI. Verify that the app doesn't crash and that LOIs are not visible in map.

**Comment by sufyanAbbasi** [comment]
I tried to delete jobs in a few ways and verified that none of them break the app. Sometimes the old LOIs/data show up momentarily while they are being resynced. It's also possible to push data to deleted jobs while the app still has them in memory, which do get stored.

After syncing, the deleted job LOIs don't show up in the app, even when the LOIs are still present in the Firestore db. I also tried deleting the jobs directly from FireStore and the app seemed to be pretty robust. 

- [x] Verified deleting jobs from the web UI
- [x] Verified deleting job IDs directly from the JSON config.
- [x] Verified deleting jobs directly from Firestore.
